packages/tlinks/src/ext/twitter.tsx

```diff
@@ -1,372 +1,432 @@
-  node: Element,
-  config: ActionAdapter,
-  callbacks: Partial<ActionCallbacksConfig>,
-  options: NormalizedObserverOptions,
+	node: Element,
+	config: ActionAdapter,
+	callbacks: Partial<ActionCallbacksConfig>,
+	options: NormalizedObserverOptions,
 ) {
-  const element = node as Element;
-  // first quick filtration
-  if (!element || element.localName !== 'div') {
-    return;
-  }
-
-  let anchor;
-
-  const linkPreview = findLinkPreview(element);
-
-  let container = findContainerInTweet(
-    linkPreview?.card ?? element,
-    Boolean(linkPreview),
-  );
-  if (linkPreview) {
-    anchor = linkPreview.anchor;
-    container && container.remove();
-    container = linkPreview.card.parentElement as HTMLElement;
-  } else {
-    if (container) {
-      return;
-    }
-    const link = findLastLinkInText(element);
-    if (link) {
-      anchor = link.anchor;
-      container = getContainerForLink(link.tweetText);
-    }
-  }
-
-  if (!anchor || !container) return;
-
-  const shortenedUrl = anchor.href;
-  const actionUrl = await resolveTwitterShortenedUrl(shortenedUrl);
-  const interstitialData = isInterstitial(actionUrl);
-
-  let actionApiUrl: string | null;
-
-  if (isTokenScriptViewerUrl(actionUrl)) {
-    // handle TokenScript viewer url
-    actionApiUrl = actionUrl.toString();
-  } else if (interstitialData.isInterstitial) {
-    const interstitialState = getExtendedInterstitialState(
-      actionUrl.toString(),
-    );
-
-    if (
-      !checkSecurity(interstitialState, options.securityLevel.interstitials)
-    ) {
-      return;
-    }
+	const element = node as Element;
+	// first quick filtration
+	if (!element || element.localName !== 'div') {
+		return;
+	}
+
+	let anchor;
+
+	const linkPreview = findLinkPreview(element);
+
+	let container = findContainerInTweet(
+		linkPreview?.card ?? element,
+		Boolean(linkPreview),
+	);
+	if (linkPreview) {
+		anchor = linkPreview.anchor;
+		container && container.remove();
+		container = linkPreview.card.parentElement as HTMLElement;
+	} else {
+		if (container) {
+			return;
+		}
+		const link = findLastLinkInText(element);
+		if (link) {
+			anchor = link.anchor;
+			container = getContainerForLink(link.tweetText);
+		}
+	}
+
+	if (!anchor || !container) return;
+
+	const shortenedUrl = anchor.href;
+	const actionUrl = await resolveTwitterShortenedUrl(shortenedUrl);
+	const interstitialData = isInterstitial(actionUrl);
+
+	let actionApiUrl: string | null;
+
+	if (isTokenScriptViewerUrl(actionUrl)) {
+		// handle TokenScript viewer url
+		actionApiUrl = actionUrl.toString();
+	} else if(isFarcasterFrameUrl(actionUrl)){
+        const chain = Number(actionUrl.searchParams.get('chain')) || 0;
+        addMargin(container).replaceChildren(
+            await createFarcasterFrame({chain, scriptURI:actionUrl.href})
+        );
+        return 
+    }else if (interstitialData.isInterstitial) {
+		const interstitialState = getExtendedInterstitialState(
+			actionUrl.toString(),
+		);
+
```

Comment:
Please only add support for the Farcaster frame, but do not change any current behavior.

![CleanShot 2024-11-15 at 10 10 51@2x](https://github.com/user-attachments/assets/32c0af88-75bd-4202-91d4-06d6afebf8b6)